src/frontend/src/optimizer/rule/distinct_agg_rule.rs

```diff
@@ -239,7 +245,7 @@ impl DistinctAggRule {
             // append `flag`.
             group_keys.insert(project.schema().len() - 1);
         }
-        Agg::new(agg_calls, group_keys, project).with_enable_two_phase(false)
+        Agg::new(agg_calls, group_keys, project).with_enable_two_phase(enable_two_phase)
```

Comment:
It brings performance regression in nexmark q15 before https://github.com/risingwavelabs/risingwave/issues/16179#issuecomment-2042049835 
so I will turn it off back.
https://risingwave-labs.slack.com/archives/C0543LNH0KU/p1712469671285099